codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -857,9 +866,16 @@ public Http2Stream closeRemoteSide() {
         }
 
         @Override
-        public int nextStreamId() {
-            // For manually created client-side streams, 1 is reserved for HTTP upgrade, so start at 3.
-            return nextStreamId > 1 ? nextStreamId : nextStreamId + 2;
+        public int incrementAndGetNextStreamId() {
```

Comment:
@nmittler - I'm not sure we can get away with 1 variable. The issue is the "generated ids" may be ahead of the "expected ids" in the event of queuing. The "expected id" is required to enforce protocol restrictions and can not progress "into the future" like the "generated ids" can. These variables may be the same if the "incrementAndGetNextStreamId" method is not used because they are kept in sync such that the "generated id" is not allowed to fall behind the "expected id".